File: sdk/ts/src/context/env.ts
/**
 * Env var helpers — mirrors config.Env / config.EnvRequired in plugin-sdk-go.
 */

/** Env helper interface exposed in PluginContext. */
export interface EnvHelper {
  /**
   * get returns the env var value or the default.
   * Missing + no default → undefined.
   */
  get(key: string, defaultValue?: string): string | undefined

  /**
   * require asserts that all named env vars are non-empty.
   * Throws on first missing var — fail-fast at install/start time.
   */
  require(keys: string[]): void

  /** getInt returns the env var parsed as an integer, or the default. */
  getInt(key: string, defaultValue: number): number

  /** getBool returns the env var parsed as boolean ("1"/"true"/"yes" → true). */
  getBool(key: string, defaultValue: boolean): boolean

  /** getList splits the env var on commas and trims whitespace. */
  getList(key: string): string[]
}

export function createEnvHelper(): EnvHelper {
  return {
    get(key: string, defaultValue?: string): string | undefined {
      const v = process.env[key]
      if (v !== undefined && v !== '') return v
      return defaultValue
    },

    require(keys: string[]): void {
      for (const key of keys) {
        if (!process.env[key] || process.env[key] === '') {
          throw new Error(`plugin SDK: required env var ${key} is not set`)
        }
      }
    },

    getInt(key: string, defaultValue: number): number {
      const v = process.env[key]
      if (!v) return defaultValue
      const n = parseInt(v, 10)
      return isNaN(n) ? defaultValue : n
    },

    getBool(key: string, defaultValue: boolean): boolean {
      const v = (process.env[key] ?? '').toLowerCase()
      if (['true', '1', 'yes', 'y', 'on'].includes(v)) return true
      if (['false', '0', 'no', 'n', 'off'].includes(v)) return false
      return defaultValue
    },

    getList(key: string): string[] {
      const v = process.env[key]
      if (!v) return []
      return v
        .split(',')
        .map((s) => s.trim())
        .filter(Boolean)
    },
  }
}

File: sdk/py/nself_plugin/context/env.py
"""
Env var helpers — mirrors config.Env / config.EnvRequired in plugin-sdk-go.
"""

from __future__ import annotations

import os
from typing import Optional


class EnvHelper:
    """
    Env var helper exposed in PluginContext.
    Mirrors config.Env, config.EnvRequired, etc. in plugin-sdk-go.
    """

    def get(self, key: str, default: Optional[str] = None) -> Optional[str]:
        """Return the env var value or default."""
        v = os.environ.get(key)
        if v is not None and v != "":
            return v
        return default

    def require(self, keys: list[str]) -> None:
        """
        Assert that all named env vars are non-empty.
        Raises ValueError on first missing var — fail-fast at install/start time.
        """
        for key in keys:
            if not os.environ.get(key):
                raise ValueError(f"plugin SDK: required env var {key} is not set")

    def get_int(self, key: str, default: int) -> int:
        """Return the env var parsed as an integer, or default."""
        v = os.environ.get(key)
        if not v:
            return default
        try:
            return int(v)
        except ValueError:
            return default

    def get_bool(self, key: str, default: bool) -> bool:
        """Return the env var parsed as boolean."""
        v = (os.environ.get(key) or "").lower()
        if v in ("true", "1", "yes", "y", "on"):
            return True
        if v in ("false", "0", "no", "n", "off"):
            return False
        return default

    def get_list(self, key: str) -> list[str]:
        """Split the env var on commas and trim whitespace. Empty → []."""
        v = os.environ.get(key, "")
        if not v:
            return []
        return [s.strip() for s in v.split(",") if s.strip()]
